buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -695,8 +695,11 @@ class BuildPlugin implements Plugin<Project> {
             // we put all our distributable files under distributions
             jarTask.destinationDir = new File(project.buildDir, 'distributions')
             project.plugins.withType(ShadowPlugin).whenPluginAdded {
-                // ensure the original jar task places its output in 'libs' so we don't overwrite it with the shadowjar
-                if (jarTask instanceof ShadowJar == false) {
+                /*
+                 * Ensure the original jar task places its output in 'libs' so that we don't overwrite it with the shadow jar. We only do
+                 * this for tasks named jar to exclude javadoc and sources jars.
+                 */
+                if (jarTask instanceof ShadowJar == false && jarTask.name == "jar") {
```

Comment:
FYI, I prefer we use the available Gradle convention constants for these kinds of things going forward. So in this case we could have used `JavaPlugin.JAR_TASK_NAME`.

Reply:
I pushed 1db38644cc784c18d0dcefc53747c75bfe7bb942.